[Swift in WebKit] Enable module verification for bmalloc

rr-codes opened the pull request:
#### e0018943a7069eb988e9668713a3d46c651455a6
<pre>
[Swift in WebKit] Enable module verification for bmalloc
<a href="https://bugs.webkit.org/show_bug.cgi?id=304186">https://bugs.webkit.org/show_bug.cgi?id=304186</a>
<a href="https://rdar.apple.com/166544086">rdar://166544086</a>

Reviewed by Aditya Keerthi and Elliott Williams.

Turn on the library module verifier for bmalloc.

This reveals several issues, so fix those too:

- The module map was malformed because it only included a subset of exposed headers and led to non-modular includes
- Add missing header includes in files that were missing some
- Use C++2b as the language standard for verification since old compilers don&apos;t recognize C++23

* Source/bmalloc/Configurations/bmalloc.xcconfig:
* Source/bmalloc/Configurations/module.modulemap:
* Source/bmalloc/bmalloc/AllocationCounts.h:
* Source/bmalloc/bmalloc/BPlatform.h:
* Source/bmalloc/bmalloc/DeferredDecommit.h:
* Source/bmalloc/bmalloc/DeferredDecommitInlines.h:
* Source/bmalloc/bmalloc/DeferredTrigger.h:
* Source/bmalloc/bmalloc/DeferredTriggerInlines.h:
* Source/bmalloc/bmalloc/EligibilityResult.h:
* Source/bmalloc/bmalloc/EligibilityResultInlines.h:
* Source/bmalloc/bmalloc/FreeList.h:
* Source/bmalloc/bmalloc/FreeListInlines.h:
* Source/bmalloc/bmalloc/IsoAllocator.h:
* Source/bmalloc/bmalloc/IsoAllocatorInlines.h:
* Source/bmalloc/bmalloc/IsoConfig.h:
* Source/bmalloc/bmalloc/IsoDeallocator.h:
* Source/bmalloc/bmalloc/IsoDeallocatorInlines.h:
* Source/bmalloc/bmalloc/IsoDirectory.h:
* Source/bmalloc/bmalloc/IsoDirectoryInlines.h:
* Source/bmalloc/bmalloc/IsoDirectoryPage.h:
* Source/bmalloc/bmalloc/IsoDirectoryPageInlines.h:
* Source/bmalloc/bmalloc/IsoHeap.h:
* Source/bmalloc/bmalloc/IsoHeapImpl.h:
* Source/bmalloc/bmalloc/IsoHeapImplInlines.h:
* Source/bmalloc/bmalloc/IsoHeapInlines.h:
* Source/bmalloc/bmalloc/IsoMallocFallback.h:
* Source/bmalloc/bmalloc/IsoPage.h:
* Source/bmalloc/bmalloc/IsoPageInlines.h:
* Source/bmalloc/bmalloc/IsoPageTrigger.h:
* Source/bmalloc/bmalloc/IsoSharedConfig.h:
* Source/bmalloc/bmalloc/IsoSharedHeap.h:
* Source/bmalloc/bmalloc/IsoSharedHeapInlines.h:
* Source/bmalloc/bmalloc/IsoSharedPage.h:
* Source/bmalloc/bmalloc/IsoSharedPageInlines.h:
* Source/bmalloc/bmalloc/IsoTLS.h:
* Source/bmalloc/bmalloc/IsoTLSAllocatorEntry.h:
* Source/bmalloc/bmalloc/IsoTLSAllocatorEntryInlines.h:
* Source/bmalloc/bmalloc/IsoTLSDeallocatorEntry.h:
* Source/bmalloc/bmalloc/IsoTLSDeallocatorEntryInlines.h:
* Source/bmalloc/bmalloc/IsoTLSEntry.h:
* Source/bmalloc/bmalloc/IsoTLSEntryInlines.h:
* Source/bmalloc/bmalloc/IsoTLSInlines.h:
* Source/bmalloc/bmalloc/IsoTLSLayout.h:
* Source/bmalloc/bmalloc/Map.h:
* Source/bmalloc/bmalloc/Object.h:
* Source/bmalloc/bmalloc/Range.h:
* Source/bmalloc/bmalloc/TZoneHeapInlines.h:
* Source/bmalloc/bmalloc/TZoneHeapManager.h:
* Source/bmalloc/bmalloc/TZoneLog.h:
* Source/bmalloc/libpas/src/libpas/pas_config.h:
* Source/bmalloc/libpas/src/libpas/pas_heap_config_utils.h:
* Source/bmalloc/libpas/src/libpas/pas_heap_ref_prefix.h:
* Source/bmalloc/libpas/src/libpas/pas_mte_config.h:
* Source/bmalloc/libpas/src/libpas/pas_segregated_page.h:
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_handle_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_segregated_view_allocator_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_small_large_map_entry.h:
* Source/bmalloc/libpas/src/libpas/pas_thread.h:
* Source/bmalloc/libpas/src/libpas/pas_try_allocate_array.h:
* Source/bmalloc/libpas/src/libpas/pas_try_allocate_primitive.h:
* Source/bmalloc/libpas/src/libpas/pas_utils.h:
* Source/bmalloc/libpas/src/libpas/pas_utils_prefix.h:

Canonical link: <a href="https://commits.webkit.org/304592@main">https://commits.webkit.org/304592@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/043bfbac2029eaf8a3f7c6e6d0f5b83bf17ad494

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/135997 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/8351 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/47273 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/143698 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/88977 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/92ea49dc-ca67-4f76-8182-15c342f899c2) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/9019 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/8198 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/103931 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/88977 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/bc129e3e-4e50-4adc-b9a7-89af7e7e6fcc) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/138943 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/6544 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/121874 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/84808 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/27336b03-5c33-42e8-a418-7fd5ab54e72b) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/6237 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/3871 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/4300 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/127957 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/115506 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/40065 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/146450 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/134480 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/8036 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/40633 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/112289 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/8055 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/6745 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/112682 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28604 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/6137 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/118175 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/61979 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/8084 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/36241 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/167267 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/7805 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/71641 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/43650 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/8026 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/7885 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/bmalloc/Configurations/bmalloc.xcconfig:
```diff
@@ -109,8 +109,8 @@ SWIFT_ACTIVE_COMPILATION_CONDITIONS_YES = BMALLOC_SWIFT_CXX_INTEROP;
+ENABLE_WK_LIBRARY_MODULE_VERIFIER = $(WK_BMALLOC_HAS_SWIFT_MODULE);
 MODULE_VERIFIER_SUPPORTED_LANGUAGES = c++ objective-c++;
-MODULE_VERIFIER_SUPPORTED_LANGUAGE_STANDARDS = c++23;
+MODULE_VERIFIER_SUPPORTED_LANGUAGE_STANDARDS = c++2b;
```
Maybe explain this in the commit message?

emw-apple reviewed (commented):
The required `BPlatform.h` includes are unfortunate, but I can't think of another way to achieve this. You may need to think of ways to convince the rest of the project that this is fine.

rr-codes committed:
[Swift in WebKit] Enable module verification for bmalloc
https://bugs.webkit.org/show_bug.cgi?id=304186
rdar://166544086

Reviewed by Aditya Keerthi and Elliott Williams.

Turn on the library module verifier for bmalloc.

This reveals several issues, so fix those too:

- The module map was malformed because it only included a subset of exposed headers and led to non-modular includes
- Add missing header includes in files that were missing some
- Use C++2b as the language standard for verification since old compilers don't recognize C++23

* Source/bmalloc/Configurations/bmalloc.xcconfig:
* Source/bmalloc/Configurations/module.modulemap:
* Source/bmalloc/bmalloc/AllocationCounts.h:
* Source/bmalloc/bmalloc/BPlatform.h:
* Source/bmalloc/bmalloc/DeferredDecommit.h:
* Source/bmalloc/bmalloc/DeferredDecommitInlines.h:
* Source/bmalloc/bmalloc/DeferredTrigger.h:
* Source/bmalloc/bmalloc/DeferredTriggerInlines.h:
* Source/bmalloc/bmalloc/EligibilityResult.h:
* Source/bmalloc/bmalloc/EligibilityResultInlines.h:
* Source/bmalloc/bmalloc/FreeList.h:
* Source/bmalloc/bmalloc/FreeListInlines.h:
* Source/bmalloc/bmalloc/IsoAllocator.h:
* Source/bmalloc/bmalloc/IsoAllocatorInlines.h:
* Source/bmalloc/bmalloc/IsoConfig.h:
* Source/bmalloc/bmalloc/IsoDeallocator.h:
* Source/bmalloc/bmalloc/IsoDeallocatorInlines.h:
* Source/bmalloc/bmalloc/IsoDirectory.h:
* Source/bmalloc/bmalloc/IsoDirectoryInlines.h:
* Source/bmalloc/bmalloc/IsoDirectoryPage.h:
* Source/bmalloc/bmalloc/IsoDirectoryPageInlines.h:
* Source/bmalloc/bmalloc/IsoHeap.h:
* Source/bmalloc/bmalloc/IsoHeapImpl.h:
* Source/bmalloc/bmalloc/IsoHeapImplInlines.h:
* Source/bmalloc/bmalloc/IsoHeapInlines.h:
* Source/bmalloc/bmalloc/IsoMallocFallback.h:
* Source/bmalloc/bmalloc/IsoPage.h:
* Source/bmalloc/bmalloc/IsoPageInlines.h:
* Source/bmalloc/bmalloc/IsoPageTrigger.h:
* Source/bmalloc/bmalloc/IsoSharedConfig.h:
* Source/bmalloc/bmalloc/IsoSharedHeap.h:
* Source/bmalloc/bmalloc/IsoSharedHeapInlines.h:
* Source/bmalloc/bmalloc/IsoSharedPage.h:
* Source/bmalloc/bmalloc/IsoSharedPageInlines.h:
* Source/bmalloc/bmalloc/IsoTLS.h:
* Source/bmalloc/bmalloc/IsoTLSAllocatorEntry.h:
* Source/bmalloc/bmalloc/IsoTLSAllocatorEntryInlines.h:
* Source/bmalloc/bmalloc/IsoTLSDeallocatorEntry.h:
* Source/bmalloc/bmalloc/IsoTLSDeallocatorEntryInlines.h:
* Source/bmalloc/bmalloc/IsoTLSEntry.h:
* Source/bmalloc/bmalloc/IsoTLSEntryInlines.h:
* Source/bmalloc/bmalloc/IsoTLSInlines.h:
* Source/bmalloc/bmalloc/IsoTLSLayout.h:
* Source/bmalloc/bmalloc/Map.h:
* Source/bmalloc/bmalloc/Object.h:
* Source/bmalloc/bmalloc/Range.h:
* Source/bmalloc/bmalloc/TZoneHeapInlines.h:
* Source/bmalloc/bmalloc/TZoneHeapManager.h:
* Source/bmalloc/bmalloc/TZoneLog.h:
* Source/bmalloc/libpas/src/libpas/pas_config.h:
* Source/bmalloc/libpas/src/libpas/pas_heap_config_utils.h:
* Source/bmalloc/libpas/src/libpas/pas_heap_ref_prefix.h:
* Source/bmalloc/libpas/src/libpas/pas_mte_config.h:
* Source/bmalloc/libpas/src/libpas/pas_segregated_page.h:
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_handle_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_segregated_view_allocator_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_small_large_map_entry.h:
* Source/bmalloc/libpas/src/libpas/pas_thread.h:
* Source/bmalloc/libpas/src/libpas/pas_try_allocate_array.h:
* Source/bmalloc/libpas/src/libpas/pas_try_allocate_primitive.h:
* Source/bmalloc/libpas/src/libpas/pas_utils.h:
* Source/bmalloc/libpas/src/libpas/pas_utils_prefix.h:

Canonical link: https://commits.webkit.org/304592@main
